tests/testutil/service.go

```diff
@@ -202,6 +203,98 @@ func (ts *TestService) GetRunningPods() string {
 	return out
 }
 
+// GetEndpointURL returns the URL for this service's app if the App object is versioned it returns
+// the versioned url.
+func (ts *TestService) GetEndpointURL() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	url, err := ts.waitForEndpointDNS()
+	if err != nil {
+		ts.T.Fatalf("Failed to get the endpoint url:  %v", err.Error())
+		return ""
+	}
+	return url
+}
+
+// GetKubeEndpointUrlByRevision returns the versioned endpoint URL by using the revision provided
+// and returns it as string
+func (ts *TestService) GetKubeEndpointURL() string {
+	err := ts.reloadApp()
```

Comment:
Believe you'll have to have both of these kube methods use wait helpers also, since DNS can take a while to come up.